src/JobsManager.cpp

```diff
@@ -0,0 +1,47 @@
+#include "JobsManager.h"
+#include <iostream>
+#include <sys/wait.h>
+
+
+/*
+* show the jobs that are currently running.
+* display the PID, command and status of each job.
+* kill jobs that have finished running.
+*/
+void JobsManager::showBackgroundJobs() const
+{
+    for (const auto& job : m_jobs) {
+        std::cout << "PID: " << job.first << ", Command: "
+            << job.second << ", Status: Running\n";
+    }
+}
+
+
+/*
+* clean up the jobs that have finished running.
+*/
+void JobsManager::cleanUpBackgroundJobs()
+{
+    for (auto it = m_jobs.cbegin(); it != m_jobs.cend();)
+    {
+        pid_t result = waitpid(it->first, nullptr, WNOHANG);
+        if (result > 0) {
+            m_jobs.erase(it++);
```

Comment:
I think using `it++` on an iterator you just erased might cause undefined behavior (I'm guessing it depends on the compiler). Instead you should use the return value of the `erase` method, like in the example here:
https://en.cppreference.com/w/cpp/container/map/erase
(I know they use a for loop without an increment in the example but it's still dangerous and a while loop should be used)